src/features/applicationView/offerCard/sendInvoiceForm/__tests__/SendInvoiceForm.test.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { act } from 'react-dom/test-utils';
+
+import SendInvoiceForm, { SendInvoiceFormProps } from '../SendInvoiceForm';
+
+const mockProps: SendInvoiceFormProps = {
+  email: 'test@example.com',
+  isSubmitting: false,
+  onCancel: jest.fn(),
+  onSubmit: jest.fn(),
+};
+
+const realDateNow = Date.now.bind(global.Date);
+
+beforeAll(() => {
+  global.Date.now = jest.fn(() => new Date('2020-09-23T00:00:00.000Z').valueOf());
+});
+
+afterAll(() => {
+  global.Date.now = realDateNow;
+});
+
+describe('SendInvoiceForm', () => {
+  const getWrapper = (props?: Partial<SendInvoiceFormProps>) => mount(<SendInvoiceForm {...mockProps} {...props} />);
+
+  it('renders normally', () => {
+    const wrapper = getWrapper();
+    expect(wrapper.render()).toMatchSnapshot();
+  });
+
+  it('shows an error if email is missing', () => {
+    const wrapper = getWrapper({ email: null });
+    expect(wrapper.find('Text.missingEmail')).toMatchInlineSnapshot(`
+      <Text
+        className="missingEmail"
+        color="critical"
+      >
+        <span
+          className="text critical span missingEmail"
+        >
+          Sähköpostiosoite puuttuu
+        </span>
+      </Text>
+    `);
+  });
+
+  it('calls onSubmit on submit', async () => {
+    const onSubmit = jest.fn();
+    const wrapper = getWrapper({ onSubmit });
+    await act(async () => {
+      wrapper.find('Form').simulate('submit');
+    });
+    expect(onSubmit).toHaveBeenCalledWith({ dueDate: new Date('2020-09-23T00:00:00.000Z') });
+  });
+
+  it('disables submit button if email is missing', async () => {
+    const wrapper = getWrapper({ email: null });
+    expect(wrapper.find('.formActionButtons').find('Button').at(1).prop('disabled')).toEqual(true);
```

Comment:
I would strongly suggest looking into testing library. Here .formActionsButtons and Button is an implementation detail and sammlest refactors could easily break these tests.

This line would be:

`expect(screen.getByRole('button', { name: 'Lähetä' })).toBeDisabled();`

Reply:
btw this is just an example, not expecting any changes to this pull request :)

Reply:
I agree, it would be great to start using Testing Library. We're a bit behind on the testing front in general, but the current schedule is a bit tight. We could make a ticket about this to work on it later

Reply:
Yeah I understand. Would be great to have it here in the future ⭐